model/hardware/registry.yaml

```
@@ -107,3 +108,37 @@ groups:
         stability: development
         brief: >
           The current state of the component
+      - id: hw.vendor.id
+        type: string
+        stability: development
+        brief: >
+          A fixed id which identifies the vendor.
+        note: >
+          if the hardware component doesn't provide it, it should default to the name.
+        examples: ['GenuineIntel', 'Seagate', 'Western Digital']
+      - id: hw.vendor.name
+        type: string
+        stability: development
+        brief: >
+          Vendor name
+        examples: ['Intel', 'Seagate', 'Western Digital']
+      - id: hw.model.id
+        type: string
+        stability: development
+        brief: >
+          A fixed id which identifies the model.
+        note: >
+          If not known should default to the name
+        examples: ['Barracuda 7200', '9000/778/B180L']
+      - id: hw.model.name
+        type: string
+        stability: development
+        brief: >
+          Descriptive model name
+        examples: ['Barracuda 7200', '11th Gen Intel(R) Core(TM) i7-1185G7 @ 3.00GHz']
+      - id: hw.serial
```

```suggestion
      - id: hw.serial_number
```

`hw.serial` is not descriptive enough. One could think it refers to a serial port, for example. `serial_number` is broadly used in SDKs, APIs, etc. I recommend using `hw.serial_number`.

Later we may add `hw.part_number` similarly.